### PR TITLE
fix(combobox): prevent spacebar from opening the menu while focusing on `chip`'s close button

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #6777

## Summary
Prevent spacebar from opening the menu while focusing on chip's close button.

### CODE REVIEW

File: packages/calcite-components/src/components.d.ts [hunk 1]
```diff
@@ -674,7 +674,7 @@ export namespace Components {
          */
         "disabled": boolean;
         /**
-          * Specifies the optional new name of the file after it is downloaded.
+          * Prompts the user to save the linked URL instead of navigating to it. Can be used with or without a value: Without a value, the browser will suggest a filename/extension See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download.
```
Comment (macandcheese) [inline packages/calcite-components/src/components.d.ts:677]: Can we remove these unrelated changes?

File: packages/calcite-components/src/components/chip/chip.tsx [hunk 2]
```diff
@@ -221,6 +221,7 @@ export class Chip
     if (event.target === this.el) {
       switch (event.key) {
         case " ":
+          event.stopPropagation();
```
Comment (macandcheese) [inline packages/calcite-components/src/components/chip/chip.tsx:224]: Is it possible to handle this within the Combobox? Maybe in `keyDownHandler` by checking to see if the event.target is not a Chip, or is only the input element, etc.?

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 3]
```diff
@@ -1918,4 +1918,58 @@ describe("calcite-combobox", () => {
       });
     });
   });
+
+  it("should not open combobox menu with spacebar while focusing on chip's close button", async () => {
```
Comment (macandcheese) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:1922]: 🥇

**Review by macandcheese — CHANGES_REQUESTED** [review]
Can we handle this in the Combobox component if possible?

File: packages/calcite-components/src/components.d.ts (hunk 1 above)
Comment (Elijbet) [reply packages/calcite-components/src/components.d.ts:677]: I think components.d.ts is supposed to be committed.

**Review by alisonailea — APPROVED** [review]

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 4]
```diff
@@ -1918,4 +1918,58 @@ describe("calcite-combobox", () => {
...
+        </calcite-combobox-item-group>
+      </calcite-combobox>
+    `);
+
+    const combobox = await page.find("calcite-combobox");
+    const openEvent = page.waitForEvent("calciteComboboxOpen");
+    await combobox.click();
+    await openEvent;
+
+    const item1 = await combobox.find("calcite-combobox-item[value=Pikachu]");
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:1949]: ⚡

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 5]
```diff
@@ -1918,4 +1918,58 @@ describe("calcite-combobox", () => {
...
+    await item1.click();
+    const item2 = await combobox.find("calcite-combobox-item[value=Charizard]");
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:1951]: 🔥

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 6]
```diff
@@ -1918,4 +1918,58 @@ describe("calcite-combobox", () => {
...
+    await item2.click();
+    const item3 = await combobox.find("calcite-combobox-item[value=Squirtle3]");
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:1953]: 💧

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 7]
```diff
@@ -1918,4 +1918,58 @@ describe("calcite-combobox", () => {
...
+    await item3.click();
+
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:1955]: No Bulbasaur love? 😭

File: packages/calcite-components/src/components/combobox/combobox.tsx [hunk 8]
```diff
@@ -693,6 +693,10 @@ export class Combobox
           }
           event.preventDefault();
         }
+
+        if ((event.target as HTMLElement).closest("calcite-chip")) {
+          event.stopPropagation();
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.tsx:698]: Can you use event cancelation instead? Per [our guidelines](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/conventions/README.md#privateinternal-events), we don't want to stop propagation of _public_ events.

**Review by jcfranco — CHANGES_REQUESTED** [review]

**Review by jcfranco — APPROVED** [review]
🎉

**Review by macandcheese — APPROVED** [review]
